tests/integration-tests/src/appinventory.rs

```diff
@@ -0,0 +1,204 @@
+    content: Vec<ContentView>,
+}
+
+impl PartialEq for ContentView {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name
+            && self.architecture == other.architecture
+            && self.version == other.version
+            && self.release == other.release
+            && self.publisher == other.publisher
+            && self.epoch == other.epoch
+    }
+}
+
+#[tokio::test]
+async fn test_twoliter_application_inventory() {
+    // We clone the latest release bob
+    let bob_version = find_latest_version("bottlerocket").await;
+    let bob_src = create_bob(bob_version.as_str()).await;
+
+    // Now we want to first get the last released twoliter so we can build our reference inventory file
+    let (_tmp_dir, latest_twoliter) = install_twoliter_from_release().await;
+    // Build and get the application inventory file contents with this twoliter
+    let reference_aif = build_and_fetch_application_inventory(&latest_twoliter, bob_src.path());
+    let reference_set: InventoryView = serde_json::from_str(reference_aif.as_str())
+        .expect("failed to deserialize reference inventory file");
+    // Now we want to build with the twoliter we built for this test (head)
+    let current_aif = build_and_fetch_application_inventory(TWOLITER_PATH, bob_src.path());
+    let current_set: InventoryView = serde_json::from_str(current_aif.as_str())
+        .expect("failed to deserialize current inventory file");
+    assert_eq!(
+        reference_set, current_set,
+        "twoliter generated different application inventory than last released twoliter"
+    );
+}
+
+fn build_and_fetch_application_inventory(
+    twoliter: impl AsRef<Path>,
+    project: impl AsRef<Path>,
+) -> String {
+    let tmp_dir = copy_project_to_temp_dir(project);
+    let twoliter_path = twoliter.as_ref().to_str().unwrap();
+    let project_path = tmp_dir.path().join("Twoliter.toml");
+    let project_path = project_path.to_str().unwrap();
+    let cmd_env = [];
+    // Twoliter update
+    let output = run_command(
+        twoliter_path,
+        ["update", "--project-path", project_path],
+        cmd_env,
+    );
+    assert!(output.status.success(), "failed to run twoliter update");
+    // Twoliter fetch
+    let output = run_command(
+        twoliter_path,
+        ["fetch", "--project-path", project_path],
+        cmd_env,
+    );
+    assert!(output.status.success(), "failed to run twoliter fetch");
+    // Build the variant
+    let output = run_command(
+        twoliter_path,
+        [
+            "build",
+            "variant",
+            "aws-ecs-2",
+            "--project-path",
+            project_path,
+        ],
+        cmd_env,
+    );
+    assert!(
+        output.status.success(),
+        "failed to run twoliter build variant"
+    );
+    // Now we should have the application inventory file
+    let aif = tmp_dir.path().join(format!(
+        "build/images/x86_64-aws-ecs-2/latest/application-inventory.json",
```

Comment:
nits:
* Use a `CONSTANT` for the target variant
* I think `x86_64` is correct since you've mentioned in person that it's `twoliter`'s default if no arch is specified, but we should probably explicitly provide an ARCH and then check for it here. This would protect us against an architecture-specific test breakage if we modify the default arch selection behavior.